shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1964,36 +1992,29 @@ @interface FlutterTextInputPlugin ()
 
 @implementation FlutterTextInputPlugin {
   NSTimer* _enableFlutterTextInputViewAccessibilityTimer;
+  std::unique_ptr<fml::WeakPtrFactory<FlutterTextInputPlugin>> _weakFactory;
+  id<FlutterTextInputDelegate> _textInputDelegate;
 }
 
-@synthesize textInputDelegate = _textInputDelegate;
-
-- (instancetype)init {
+- (instancetype)initWithDelegate:(id<FlutterTextInputDelegate>)textInputDelegate {
   self = [super init];
 
   if (self) {
+    _textInputDelegate = textInputDelegate;
```

Comment:
You should add a comment as to why it's safe to have a weak reference here.

Reply:
Add a comment for `textInputDelegate`. Is that what you meant (or you meant a comment for `_weakFactory`)?

Reply:
I was thinking something like:
```objc
// `_textInputDelegate` is a weak reference because the it should retain FlutterTextInputPlugin.
```

Reply:
Done. Does the method -> property change look right to you?

Reply:
Yep, lgtm!